### PR TITLE
Change enriched git index name, add functionality to choose the index to be analysed

### LINKED ISSUE #17: Make git the name of the enriched index for git
AFAIK, by default Mordred is creating the alias for the git enriched index as `git`. However, it seems Manuscripts is looking for `git_enrich` (see #16). Could somebody more knowledgeable on Manuscripts confirm this, so that we keep this issue open, or deny, and we just close it?

### PR DESCRIPTION
**Pull request by aswanipranjal**
Closes #17 and #21 

The user can choose the name of the indices to be analysed by using the `--indices` flag.

### CODE REVIEW

**Comment by aswanipranjal** [comment]
@jgbarah the tests might fail because of flake8, they will pass after #18 is merged.

**Comment by aswanipranjal** [comment]
1. So should I squash the commits? I think they are okay because they address 2 different bugs.
2. Yeah, I can probably reset and make more descriptive commit comments.
3. From the top of my head we can probably use some thing like:
```
parser.add_argument("indices", default=[], nargs='*')
```
which basically creates a list of all the indices that we want to use in the report. The problem with this will be that names of all the indices have to be present for it to work.
The second solution that I can think of is that we can have separate args for each index name? but it might get too cluttered.

I'll search more and come up with a better solution though.

**Comment by jgbarah** [comment]
I think forcing users to specify all index names, when maybe they only need to specify one because defaults are ok for the rest is a bit too much... If cannot find out something better, we could:

* have specific parameters for the index for each data source, such as `--index_git`,`--index_github` and so on, or

* having a single parameter, that has, in the same order of data sources, the list of index names to override. For example, `--data_sources git github slack --indexes git_index github_index` would change the default for the git and github indexes, but not for slack index.

I'm open to other ideas...

**Comment by aswanipranjal** [comment]
I like the second idea better. I'll try it out!

**Comment by aswanipranjal** [comment]
@jgbarah: now the users can select the name of the different indices to be analysed using the `--indices` flag. Example:
```
bin/manuscripts -u http://localhost:9200 -n MyProject --data-sources git github --indices git_index github_index -d tmp/report
```

Here, if no custom index has been provided, then the default index for that data source will be used. The only caveat being that the default index/data source has to be at the last.

File: manuscripts/report.py [hunk 1]
```diff
@@ -134,11 +134,19 @@ def __init__(self, es_url, start, end, data_dir=None, filters=None,
...
+        self.indices = indices if indices else []
         # End temporal hack
         self.config = self.__get_config(self.data_sources)
         self.report_name = report_name
         self.projects = projects
 
+        # Create an index name dict which, for each data_source, will give the
+        # name of the elasticsearch index that has to be used.
+        self.index_dict = defaultdict(lambda: None)
+        for i in range(len(self.indices)):
```
Comment (jgbarah) [inline manuscripts/report.py]: What about `for pos, index in enumerate(self.indices)` ? I find it more pythonic, and it gives you index, that you need below, in the body of the loop.

File: manuscripts/report.py [hunk 2]
```diff
@@ -134,11 +134,19 @@ def __init__(self, es_url, start, end, data_dir=None, filters=None,
...
+        # Check for custom indices
+        self.indices = indices if indices else []
         # End temporal hack
         self.config = self.__get_config(self.data_sources)
         self.report_name = report_name
         self.projects = projects
 
+        # Create an index name dict which, for each data_source, will give the
+        # name of the elasticsearch index that has to be used.
+        self.index_dict = defaultdict(lambda: None)
```
Comment (jgbarah) [inline manuscripts/report.py]: I think you can just initialize this to `{}`, I don't see a need for defaultdict. But please, let us know if you find it important for some reason.

File: manuscripts/report.py [hunk 3]
```diff
@@ -92,7 +92,7 @@ class Report():
 
     def __init__(self, es_url, start, end, data_dir=None, filters=None,
                  interval="month", offset=None, data_sources=None,
-                 report_name=None, projects=False):
+                 report_name=None, projects=False, indices=None):
```
[the remark this answers is not in the document]
Comment (aswanipranjal) [reply manuscripts/report.py]: Yeah, that makes more sense.

File: manuscripts/report.py (hunk 2 above)
Comment (aswanipranjal) [reply manuscripts/report.py]: I used a default dict because in case a key is not present, it will return a `None` for the value instead of a `KeyError`.

**Comment by aswanipranjal** [comment]
@jgbarah, could you please take a look at the code?

**Comment by aswanipranjal** [comment]
@alpgarcia, could you please review the pull request once?

**Pull request merged**